### DISCUSSION TITLE
How to auto adjust Font within the Boundary using Fabric.js textbox in Node.js?

### DISCUSSION THREAD

**Discussion by aquodigital** [fabricjs/fabric.js#7768] [Q&A] [answered]
I'm using Fabric.js object to create images dynamically in `Node.js`

How do I reduce the font size if the text is too long to make sure that the text does not exceed the blue coloured boundary.

I replace the `{{name}}` from the `Textbox` dynamically.

I'm able to wrap the text horizontally. However, it overflows vertically.

I'm using custom fonts, different fonts seem to take up different space.

How do I calculate the upper limit for font size?

Reference Image 

https://i.stack.imgur.com/f4r98.jpg

**Comment by ShaMan123** [comment, marked as answer]
You will need to calculate it yourself AFAIK
Try looking into `_getTransformedDimensions` 

Perhaps a better approach would be to scale the text down by some factor that can be drived from `_getTransformedDimensions` 
There's a util `findScaleToFit` that might come in handy
